src/main/java/org/stellar/sdk/Sep10Challenge.java

```diff
@@ -40,4 +46,366 @@ public static String newChallenge(
+      throw new InvalidSep10ChallengeException("Operation type should be ManageData.");
+    }
+    ManageDataOperation manageDataOperation = (ManageDataOperation) operation;
+
+    // verify that transaction envelope has a correct signature by server's signing key
+    String clientAccountId = manageDataOperation.getSourceAccount();
+    if (clientAccountId == null) {
+      throw new InvalidSep10ChallengeException("Operation should have a source account.");
+    }
+
+    if (manageDataOperation.getValue().length != 64) {
+      throw new InvalidSep10ChallengeException("Operation value should be a 64 bytes base64 random string.");
+    }
+
+    if (transaction.getSignatures().isEmpty()) {
+      throw new InvalidSep10ChallengeException("Transaction has no signatures.");
+    }
+
+    if (!verifyTransactionSignedByAccountId(transaction, serverAccountId)) {
+      throw new InvalidSep10ChallengeException(String.format("Transaction not signed by server: %s.", serverAccountId));
+    }
+
+    // verify that transaction sequenceNumber is equal to zero
+    if (transaction.getSequenceNumber() != 0L) {
+      throw new InvalidSep10ChallengeException("The transaction sequence number should be zero.");
+    }
+
+    return new ChallengeTransaction(transaction, clientAccountId);
+  }
+
+  /**
+   * Verifies that for a SEP 10 challenge transaction
+   * all signatures on the transaction are accounted for. A transaction is
+   * verified if it is signed by the server account, and all other signatures
+   * match a signer that has been provided as an argument. Additional signers can
+   * be provided that do not have a signature, but all signatures must be matched
+   * to a signer for verification to succeed. If verification succeeds a list of
+   * signers that were found is returned, excluding the server account ID.
+   *
+   * @param challengeTransaction SEP-0010 transaction challenge transaction in base64.
+   * @param serverAccountId      Account ID for server's account.
+   * @param network              The network to connect to for verifying and retrieving.
+   * @param signers              The signers of client account.
+   * @return a list of signers that were found is returned, excluding the server account ID.
+   * @throws InvalidSep10ChallengeException If the SEP-0010 validation fails, the exception will be thrown.
+   * @throws IOException                    If read XDR string fails, the exception will be thrown.
+   */
+  public static List<String> verifyChallengeTransactionSigners(String challengeTransaction, String serverAccountId, Network network, List<String> signers) throws InvalidSep10ChallengeException, IOException {
+    if (signers == null || signers.isEmpty()) {
+      throw new InvalidSep10ChallengeException("No signers provided.");
+    }
+
+    // Read the transaction which validates its structure.
+    ChallengeTransaction readableChallengeTransaction = readChallengeTransaction(challengeTransaction, serverAccountId, network);
```

Comment:
I prefer the variable name `parsedChallengeTransaction`

Reply:
or just `parsed`

Reply:
Or `readChallengeTransaction`? The past tense is needed though instead of the `able`.